### PR TITLE
Add idl and constructor test for coalesced events

### PR DESCRIPTION
**Pull request by NavidZ**
@dtapuska @mustaqahmed 
Some tests for w3c/pointerevents#22

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Chrome (unstable channel) #
Testing web-platform-tests at revision deb89c305c7f5118780fee09b8fd6a2008cebaf9
Using browser at version 58.0.3004.3 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4797/pointerevents/extension/pointerevent_constructor.html">/pointerevents/extension/pointerevent_constructor.html</a></summary>


|             Subtest              | Results |
|----------------------------------|---------|
|                                  | OK      |
| `PointerEvent: Constructor test` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4797/pointerevents/extension/idlharness.html">/pointerevents/extension/idlharness.html</a></summary>


|                                                  Subtest                                                  | Results |
|-----------------------------------------------------------------------------------------------------------|---------|
|                                                                                                           | OK      |
| `PointerEvent interface: operation getCoalescedEvents()`                                                  | FAIL    |
| `PointerEvent interface: existence and properties of interface object`                                    | PASS    |
| `PointerEvent interface object name`                                                                      | PASS    |
| `PointerEvent interface: existence and properties of interface prototype object's "constructor" property` | PASS    |
| `PointerEvent interface: existence and properties of interface prototype object`                          | FAIL    |
| `PointerEvent interface object length`                                                                    | PASS    |

</details>

**Comment by w3c-bots** [comment]
# Firefox (nightly channel) #
Testing web-platform-tests at revision deb89c305c7f5118780fee09b8fd6a2008cebaf9
Using browser at version BuildID 20170123125947; SourceStamp 36486fdc3813ef7943ae5b07b4128866d1938a6c
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary><a href="http://w3c-test.org/submissions/4797/pointerevents/extension/pointerevent_constructor.html">/pointerevents/extension/pointerevent_constructor.html</a></summary>


|             Subtest              | Results |
|----------------------------------|---------|
|                                  | OK      |
| `PointerEvent: Constructor test` | FAIL    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/4797/pointerevents/extension/idlharness.html">/pointerevents/extension/idlharness.html</a></summary>


|                                                  Subtest                                                  | Results |
|-----------------------------------------------------------------------------------------------------------|---------|
|                                                                                                           | OK      |
| `PointerEvent interface: operation getCoalescedEvents()`                                                  | FAIL    |
| `PointerEvent interface: existence and properties of interface object`                                    | FAIL    |
| `PointerEvent interface object name`                                                                      | FAIL    |
| `PointerEvent interface: existence and properties of interface prototype object's "constructor" property` | FAIL    |
| `PointerEvent interface: existence and properties of interface prototype object`                          | FAIL    |
| `PointerEvent interface object length`                                                                    | FAIL    |

</details>

**Comment by NavidZ** [comment]
@dtapuska

File: pointerevents/extension/pointerevent_constructor.html [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    <h1>PointerEvent: Dispatch custom event with coalesced events</h1>
+    <h4>Test Description: This test checks if PointerEvent constructor works properly with respect to the coalesced events.</h4>
+    <div id="target0"></div>
+    <script>
+        async_test(function() {
+            var target0 = document.getElementById("target0");
+            // set values for non-default constructor
+            on_event(target0, "pointermove", this.step_func(function(event) {
+                generate_tests(assert_equals, [
+                    ["getCoalescedEvents().length", event.getCoalescedEvents().length, 2],
```
Comment (dtapuska) [inline pointerevents/extension/pointerevent_constructor.html]: What about target and current target should we assert they are the same as the original event?

**Comment by dtapuska** [comment]
lgtm

**Pull request merged**